tests/nest_tests/nest_delay_based_variables_test.py

```diff
@@ -0,0 +1,199 @@
+                             suffix=self.suffix)
+        recordables = ["V_m", "U_m"]
+
+        # Run simulation with delay
+        recordable_events_delay, times = self.run_simulation(neuron_model_name,
+                                                             module_name,
+                                                             recordables,
+                                                             delay=5.0, dc_gen=True)
+
+        # Run the simulation with no delay
+        recordable_events, times = self.run_simulation(neuron_model_name,
+                                                       module_name,
+                                                       recordables,
+                                                       delay=0.0, dc_gen=True)
+
+        if TEST_PLOTS:
+            self.plot_fig(times, recordable_events_delay, recordable_events, neuron_model_name + ".png")
+
+    def test_eqns_with_delay_vars_mixed_solver(self):
+        input_path = os.path.join(os.path.realpath(os.path.join(os.path.dirname(__file__), "resources",
+                                                                "DelayBasedVariablesWithMixedSolver.nestml")))
+        neuron_model_name = "delay_variables_mixed_solver_nestml"
+        module_name = neuron_model_name + "_module"
+        generate_nest_target(input_path=input_path,
+                             target_path=self.target_path,
+                             logging_level=self.logging_level,
+                             module_name=module_name,
+                             suffix=self.suffix)
+        recordables = ["V_m", "w"]
+        spikes = [1.0, 1.0, 1.5, 1.5, 6.7, 10.0, 10.5, 10.5, 10.5, 10.5, 11.3, 11.3, 11.4, 11.4, 20., 22.5, 30.,
+                  40., 42., 42., 42., 50.5, 50.5, 75., 88., 93., 95., 96.7, 98.8]
+
+        # Simulate with delay
+        recordable_events_delay, times = self.run_simulation(neuron_model_name,
+                                                             module_name,
+                                                             recordables,
+                                                             delay=45.0, spikes=spikes)
+
+        # Simulate without delay
+        recordable_events, times = self.run_simulation(neuron_model_name,
+                                                       module_name,
+                                                       recordables,
+                                                       delay=0., spikes=spikes)
+
+        if TEST_PLOTS:
+            self.plot_fig(times, recordable_events_delay, recordable_events, neuron_model_name + ".png")
+
+    @pytest.fixture(scope="function", autouse=True)
+    def cleanup(self):
+        # Run the test
+        yield
+
+        # clean up
+        import shutil
+        if self.target_path:
+            try:
+                shutil.rmtree(self.target_path)
+            except Exception:
+                pass
```

Comment:
This test does not actually contain any assertions. It would be great if we had separate delayed and non-delayed variables, so that we could then numerically check that when we remove the delay again from the curves in Python, they overlap to floating-point precision. Or you could numerically check some simple cases of DDEs (like oscillation); see for example https://fse.studenttheses.ub.rug.nl/14274/1/Delay_Differential_Equations.pdf.

Reply:
I added simpler test cases for numeric and mixed solver cases. I have also added assert for the analytical solver case as the equations are linear and the results can be easily compared. However, I haven't found a way to assert for the numeric and mixed solvers yet. Do you have any ideas?